tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -199,6 +199,25 @@ public function testColumnCollation()
         $this->assertEquals('utf8_general_ci', $columns['bar']->getPlatformOption('collation'));
     }
 
+    public function testColumnCharset()
+    {
+        $table = new Table('test_charset');
+        $table->addOption('collate', $collation = 'latin1_swedish_ci');
+        $table->addOption('charset', 'latin1');
+        $table->addColumn('id', 'integer');
+        $table->addColumn('text', 'text');
+        $table->addColumn('foo', 'text')->setPlatformOption('charset', 'latin1');
+        $table->addColumn('bar', 'text')->setPlatformOption('charset', 'utf8');
+        $this->_sm->dropAndCreateTable($table);
+
+        $columns = $this->_sm->listTableColumns('test_charset');
+
+        $this->assertArrayNotHasKey('charset', $columns['id']->getPlatformOptions());
+        $this->assertEquals('latin1', $columns['text']->getPlatformOption('charset'));
```

Comment:
This looks wrong. The charset is set implicit here through the table's default charset. However why is it `latin1` here? As far as I can see, we set it implicitly to `utf8` [here](https://github.com/mRoca/dbal/blob/feature/mysql-charset/lib/Doctrine/DBAL/Platforms/MySqlPlatform.php#L483-L488)